Broken build on MSYS2

Commit: **f963b63afa0e057cfb9eba4d88407c6a0850a0d8** Breaks the MSYS2 build with the following messages:

`llama_util.h: In constructor 'llama_mmap::llama_mmap(llama_file*)':
llama_util.h:213:9: error: 'WIN32_MEMORY_RANGE_ENTRY' was not declared in this scope
  213 |         WIN32_MEMORY_RANGE_ENTRY range;
      |         ^~~~~~~~~~~~~~~~~~~~~~~~
llama_util.h:214:9: error: 'range' was not declared in this scope
  214 |         range.VirtualAddress = addr;
      |         ^~~~~
llama_util.h:216:14: error: 'PrefetchVirtualMemory' was not declared in this scope
  216 |         if (!PrefetchVirtualMemory(GetCurrentProcess(), 1, &range, 0)) {
      |              ^~~~~~~~~~~~~~~~~~~~~
`

--- a/examples/main/main.cpp
+++ b/examples/main/main.cpp
@@ -1,3 +1,8 @@
+// Defines sigaction on msys:
+#ifndef _GNU_SOURCE
+#define _GNU_SOURCE
+#endif
+
 #include "common.h"
 #include "llama.h"
 

--- a/ggml.c
+++ b/ggml.c
@@ -1,4 +1,4 @@
-// Defines CLOCK_MONOTONIC and asprintf on Linux
+// Defines CLOCK_MONOTONIC on Linux
 #define _GNU_SOURCE
 
 #include "ggml.h"
@@ -50,6 +50,7 @@ typedef HANDLE pthread_t;
 
 typedef DWORD thread_ret_t;
 static int pthread_create(pthread_t* out, void* unused, thread_ret_t(*func)(void*), void* arg) {
+    (void) unused;
     HANDLE handle = CreateThread(NULL, 0, (LPTHREAD_START_ROUTINE) func, arg, 0, NULL);
     if (handle == NULL)
     {
@@ -61,6 +62,7 @@ static int pthread_create(pthread_t* out, void* unused, thread_ret_t(*func)(void
 }
 
 static int pthread_join(pthread_t thread, void* unused) {
+    (void) unused;
     return (int) WaitForSingleObject(thread, INFINITE);
 }
 

--- a/llama.cpp
+++ b/llama.cpp
@@ -1,3 +1,8 @@
+// Defines fileno on msys:
+#ifndef _GNU_SOURCE
+#define _GNU_SOURCE
+#endif
+
 #include "llama_util.h"
 #include "llama.h"
 #include "llama_internal.h"

--- a/llama_util.h
+++ b/llama_util.h
@@ -26,7 +26,9 @@
 
 #if defined(_WIN32)
     #define WIN32_LEAN_AND_MEAN
-    #define NOMINMAX
+    #ifndef NOMINMAX
+        #define NOMINMAX
+    #endif
     #include <windows.h>
     #include <io.h>
     #include <stdio.h> // for _fseeki64
@@ -209,6 +211,7 @@ struct llama_mmap {
             throw format("MapViewOfFile failed: %s", llama_format_win_err(error).c_str());
         }
 
+        #if _WIN32_WINNT >= _WIN32_WINNT_WIN8
         // Advise the kernel to preload the mapped memory
         WIN32_MEMORY_RANGE_ENTRY range;
         range.VirtualAddress = addr;
@@ -217,6 +220,9 @@ struct llama_mmap {
             fprintf(stderr, "warning: PrefetchVirtualMemory failed: %s\n",
                     llama_format_win_err(GetLastError()).c_str());
         }
+        #else
+        #pragma message("warning: You are building for pre-Windows 8; prefetch not supported")
+        #endif // _WIN32_WINNT >= _WIN32_WINNT_WIN8
     }
 
     ~llama_mmap() {
@@ -338,8 +344,8 @@ struct llama_mlock {
             // Hopefully a megabyte is enough overhead:
             size_t increment = size + 1048576;
             // The minimum must be <= the maximum, so we need to increase both:
-            min_ws_size += size;
-            max_ws_size += size;
+            min_ws_size += increment;
+            max_ws_size += increment;
             if (!SetProcessWorkingSetSize(GetCurrentProcess(), min_ws_size, max_ws_size)) {
                 fprintf(stderr, "warning: SetProcessWorkingSetSize failed: %s\n",
                         llama_format_win_err(GetLastError()).c_str());

